ruby/enumerator_size.md

```diff
@@ -0,0 +1,42 @@
+# Enumerator#size
+
+We've been recently wrapping a couple of external APIs with [Enumerators](http://ruby-doc.org/core-2.2.1/Enumerator.html)
+and we were having some small issues when asking an enumerator for its
+size. So, let's say you have something like this:
+
+```ruby
+result_from_api = [1, 2, 3]
+
+wrapper = Enumerator.new do |yielder|
+  result_from_api.each do |n|
+    yielder << n
+  end
+end
+```
+
+Then, if you call `#size`, you'd get an infamous `nil`:
+
+```ruby
+  wrapper.size # => nil
```

Comment:
For consistency's sake, we should probably remove the unnecessary indentation on this line.